src/methods/join.js

```diff
@@ -635,6 +627,9 @@ export class JoinImplementation {
       return new ReturnCompletion(val, joinCondition.expressionLocation);
     }
     if (result1 instanceof ThrowCompletion && result2 instanceof ThrowCompletion) {
+      getAbstractValue = (v1: void | Value, v2: void | Value) => {
```

Comment:
We need this because "throw __empty" is used as a signal for "the normal branch". See Program.js/274.